File: 0819_most_common_word/most_common_word.js
/**
 * Time:    O(m + n)
 * Space:   O(m + n)
 * @param   {string} paragraph
 * @param   {string[]} banned
 * @return  {string}
 */
const mostCommonWord = (paragraph, banned) => {
  const s = new Set(banned)
  const h = {}
  const p = paragraph.replace(/[^A-Za-z]/g, " ").split(" ")
  for (let w of p) {
    var temp = w.toLowerCase()
    if (!s.has(temp) && temp.length) {
      if (temp in h) h[temp]++
      else h[temp] = 1
    }
  }
  let maxCount = 0
  let maxKey = ""
  for (let [key, value] of Object.entries(h)) {
    if (value > maxCount) {
      maxCount = value
      maxKey = key
    }
  }
  return maxKey
}

File: 0819_most_common_word/most_common_word.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    // Time: O(m + n)
    // Space: O(m + n)
    public String mostCommonWord(String paragraph, String[] banned) {
        Set<String> banSet = new HashSet<>(Arrays.asList(banned));
        return Arrays.asList(paragraph.replaceAll("[!?',;.]", " ")
            .toLowerCase()
            .split("\\s+"))
            .stream()
            .filter(word -> !banSet.contains(word))
            .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()))
            .entrySet()
            .stream()
            .max(Map.Entry.comparingByValue())
            .get()
            .getKey();
    }
}